### PR TITLE
Remove delay time on workflow submit

### PR DESCRIPTION
**Pull request by jtronge**
This should address issue #705. Tasks to start the GDB and initialize workflows are now offloaded to a celery backend based on Redis. I'm marking this as WIP since we still need to figure out the containers to use.

### CODE REVIEW

**Comment by jtronge** [comment]
This adds a new config option `redis_image` for the Redis container. Right now it's just using the official one from docker hub.

**Comment by jtronge** [comment]
The documentation for the new redis container should be updated now. I also changed `bee_dep_image` to `neo4j_image` in the config to be consistent.

**Comment by jtronge** [comment]
I just added a command `beeflow core pull-deps` to download both the Neo4j and Redis containers. I think this should address #702.

**Review by pagrubel — CHANGES_REQUESTED** [review]
Should the command really be ``beeflow config pull-deps``?
Also, can you use the code that is used to load the charliecloud module if it doesn't exist to load it or warn that charliecloud is not available when using the command with the url for charliecloud, otherwise the user gets an error saying ch-image does not exist.
And the command to download the containers should also be documented in the command line interface section of either core or config wherever it ends up.

This probably should have been a separate pull request. I still need to review the rest of the pull request.

**Comment by pagrubel** [comment]
I'm getting an error when I try to run the simple cat-grep-tar example on darwin. Attaching the error log. 
[bee-error-1697478790.log](https://github.com/lanl/BEE/files/12920306/bee-error-1697478790.log)

**Comment by pagrubel** [comment]
> I'm getting an error when I try to run the simple cat-grep-tar example on darwin. Attaching the error log. [bee-error-1697478790.log](https://github.com/lanl/BEE/files/12920306/bee-error-1697478790.log)

Deleted my bee_workdir to have a clean copy and this cleared up the problem.

File: beeflow/client/core.py [hunk 1]
```diff
@@ -181,6 +181,47 @@ def start_scheduler():
...
+        data_dir = 'data'
+        os.makedirs(os.path.join(paths.redis_root(), data_dir), exist_ok=True)
+        conf_name = 'redis.conf'
+        container_path = paths.redis_container()
+        # If it exists, we assume that it actually has a valid container
+        if not os.path.exists(container_path):
+            subprocess.check_call(['ch-convert', '-i', 'tar', '-o', 'dir',
+                                   bc.get('DEFAULT', 'redis_image'), container_path])
+        # Dump the config
+        with open(os.path.join(paths.redis_root(), conf_name), 'w', encoding='utf-8') as fp:
```
Comment (rstyd) [inline beeflow/client/core.py]: I think we should only dump a config if one doesn't currently exist.

File: beeflow/wf_manager/resources/wf_actions.py [hunk 2]
```diff
@@ -22,20 +22,12 @@ def __init__(self):
...
-                                         status='ok'), 200)
-            return resp
-        wfi.execute_workflow()
-        tasks = wfi.get_ready_tasks()
-        wf_utils.schedule_submit_tasks(wf_id, tasks)
-        wf_id = wfi.workflow_id
-        wf_utils.update_wf_status(wf_id, 'Running')
-        db.workflows.update_workflow_state(wf_id, 'Running')
-        resp = make_response(jsonify(msg='Started workflow!', status='ok'), 200)
+        if wf_utils.start_workflow(wf_id):
```
Comment (rstyd) [inline beeflow/wf_manager/resources/wf_actions.py:25]: This is a great change!

**Review by rstyd — APPROVED** [review]
Everything looks great to me. I approve!

**Review by pagrubel — APPROVED** [review]
Thanks Jake this is great!

**Pull request merged**